x-pack/test/defend_workflows_cypress/fleet_server.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ChildProcess, spawn } from 'child_process';
+import { ToolingLog } from '@kbn/tooling-log';
+import axios, { AxiosRequestConfig } from 'axios';
+import { Manager } from './resource_manager';
+import { getLatestVersion } from './artifact_manager';
+
+export interface AgentManagerParams {
+  user: string;
+  password: string;
+  kibanaUrl: string;
+  esHost: string;
+  esPort: string;
+}
+
+export class FleetManager extends Manager {
```

Comment:
same comment - we might be able to convert some of this code here to re-usable service: 

https://github.com/elastic/kibana/blob/d1194d488a6626731e1ee9cc43e599e1c7f48e8f/x-pack/plugins/security_solution/scripts/endpoint/endpoint_agent_runner/fleet_server.ts#L47